lib/services/generator-extensions-service.ts

```diff
@@ -0,0 +1,29 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import dependencyExtensionsServiceLib = require("./dependency-extensions-service-base");
+import path = require("path");
+import util = require("util");
+
+export class GeneratorExtensionsService extends dependencyExtensionsServiceLib.DependencyExtensionsServiceBase implements IGeneratorExtensionsService {
+	constructor($fs: IFileSystem,
+		$httpClient: Server.IHttpClient,
+		$logger: ILogger,
+		private $dependencyConfigService: IDependencyConfigService) {
+			super($fs, $httpClient, $logger);
+	}
+
+	public getGeneratorCachePath(generatorName: string, appScaffoldingPath: string): string {
+		return path.join(appScaffoldingPath, "latest", "node_modules", generatorName);
+	}
+
+	public prepareGenerator(generatorName: string, appScaffoldingPath: string): IFuture<void> {
+		return (() => {
+			var generatorConfig = this.$dependencyConfigService.getGeneratorConfig(generatorName).wait();
+			generatorConfig.pathToSave = this.getGeneratorCachePath(generatorName, appScaffoldingPath);
```

Comment:
if generatorConfig is undefined (getGeneratorConfig  is using _.find which will return undefined in case it cannot find the searched element), this call will fail